### PR TITLE
Update Autocomplete Component

### PR DESCRIPTION
**Pull request by pkrawc**
This PR updates the autocomplete component to add support for manual address adding.

The real work for manually adding and address is left up to the consumer but the API now accepts an `onManualSelection` function that should return the UI to update an address manually.

`OnManualSelection`

```jsx
(submit, [isOpen, setOpen]) => {
  return (
    <Dialog open={isOpen} toggle={() => setOpen((o) => !o)}>
      <ConfirmationDialog style={{ zIndex: "99999" }}>
        <DialogHeader title="Enter address manually" />
        <Grid>
          <Col as={Input} span={[9]} label="Street Address" />
          <Col as={Input} span={[3]} label="No." />
          <Col as={Input} span={[8]} label="City" />
          <Col as={Input} span={[4]} label="State" />
          <Col as={Input} span={[6]} label="Zip Code" />
          <Col
            as={Input}
            span={[6]}
            label="Country"
            defaultValue="United States"
          />
          <Col
            as={Button}
            sx={{ justifySelf: "end" }}
            variant="primary"
            onClick={(e) =>
              submit({
                formatted_address:
                  "1234 North Augusta Boulevard, Chicago, IL, United States",
              })
            }
          >
            submit
          </Col>
        </Grid>
      </ConfirmationDialog>
    </Dialog>
  )
}
```

### CODE REVIEW

File: packages/autocomplete/src/Autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  }
+  function handleKeyDown({ keyCode }) {
+    if (selection) {
+      setInput("")
+      return clearSelection(onClear)
+    }
+    switch (keyCode) {
+      // Tab Key
+      case 9:
+        console.log(listRef.current?.firstChild)
```
Comment (creaumond) [inline packages/autocomplete/src/Autocomplete.js:62]: Can get rid of this.

File: packages/autocomplete/src/Autocomplete.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                    "&:hover, &:focus": { bg: "blue_100" },
+                  }}
+                  p="2rem"
+                  onClick={() => handleSelect(place)}
+                >
+                  {place.description}
+                </Box>
+              )
+            })}
+            {suggestions < 5 && (
```
Comment (creaumond) [inline packages/autocomplete/src/Autocomplete.js:144]: Does this work? I'd imagine this should be `suggestions.length`

**Review by creaumond — COMMENTED** [review]
This is a great and very clean approach. Everything was simple to read through and understand. I have a few minor items of potential issues, but great job. 

I'm not suggesting you change it, but the only thing that kind of threw me off was the variable name `called` I found to be confusing / not very descriptive. You can take from that whatever you want.

File: packages/autocomplete/src/Autocomplete.js (hunk 2 above)
Comment (pkrawc) [reply packages/autocomplete/src/Autocomplete.js:144]: Lol, you're right.

**Comment by pkrawc** [comment]
> I'm not suggesting you change it, but the only thing that kind of threw me off was the variable name `called` I found to be confusing / not very descriptive. You can take from that whatever you want.

@creaumond I'm open to other suggestions. I went with `called` cause it's a key on the `useMutation` hook so it seemed kind of standard.

**Review by quaardvark — APPROVED** [review]

**Pull request merged**